### PR TITLE
Data availability layer tutorial

### PR DESCRIPTION
**Pull request by timothymcmackin**
This is my suggestion to simplify the DAL tutorial in https://github.com/trilitech/tezos-developer-docs/pull/257.

Major changes:
- Split into separate files
- Get style and such in sync with the rest of the docs
- Add prereqs and clarify steps in general
- Make sure that it works
- Ensure that the user is clearly directed to do everything that they need to do

Preview: https://docs-staging-git-dal-tutorial-trili-tech.vercel.app/tutorials/build-files-archive-with-dal

### CODE REVIEW

**Comment by lthms** [comment]
You can probably target the main branch directly no? I’ll close my own PR

File: docs/tutorials/build-files-archive-with-dal.mdx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+Smart Rollups must store the data because it is available on the DAL for only a short time.
+1. Users who know the hash of the data can download it from the Smart Rollup node.
+
+The overall workflow is summarized in the following figure:
+
+<LucidDiagram width="640px" height="480px" src="https://lucid.app/documents/embedded/cc422278-7319-4a2f-858a-a7b72e1ea3a6" id="ljs6hoejIr1H" />
+
+There are many steps in the DAL process, but the most complicated parts (storing and sharing data) are handled automatically by the various daemons in the Octez suite.
+
+:::note The Smart Rollup does not support the DAL
```
Comment (lthms) [inline docs/tutorials/build-files-archive-with-dal.mdx:146]:
```suggestion
:::note The Smart Rollup Installer does not support the DAL
```

File: docs/tutorials/build-files-archive-with-dal/using-full-slot.mdx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+1. Inspect the contents of the slot by running this command:
+
+   ```bash
+   hexdump -C slot.bin
+   ```
+
+   For example, if you uploaded a file that contained the text "Hi! This is a message to go on the DAL." the slot might look like this:
+
+   ```
+   00000000  22 34 38 36 39 32 31 32  30 35 34 36 38 36 39 37  |"486921205468697|
```
Comment (lthms) [inline docs/tutorials/build-files-archive-with-dal/using-full-slot.mdx:132]: The output of `hexdump` does not look quite right. I’ll have to retry very quickly, but ideally, you should see the text itself.

File: docs/tutorials/build-files-archive-with-dal.mdx (hunk 1 above)
Comment (timothymcmackin) [reply docs/tutorials/build-files-archive-with-dal.mdx]: ohhh, now I understand that part!

File: docs/tutorials/build-files-archive-with-dal/using-full-slot.mdx (hunk 2 above)
Comment (timothymcmackin) [reply docs/tutorials/build-files-archive-with-dal/using-full-slot.mdx]: Changed the command to `xxd -r -p slot.bin`, which prints the text.

**Comment by timothymcmackin** [comment]
Merging with approvals from Thomas and François.

**Pull request merged**